js/lib/index.js

```diff
@@ -0,0 +1,703 @@
+   *     },
+   *     // and more
+   *   ]
+   * }
+   * ```
+   *
+   * Only previously set objects using the same private key as this instance may
+   * be overwritten.
+   *
+   * @param {object} work - the Work to be set; see method doc for structure
+   *
+   * @returns {Promise<BatchStatus>} a promise for the result of transaction submission
+   *
+   * @since 1.0.0
+   */
+  setWork (work) {
+    return _submitOmiTransaction(this._sawtoothRestUrl,
+                                 this._privateKey,
+                                 'SetWork',
+                                 Work,
+                                 'title',
+                                 work)
+  }
+
+  /**
+   * Fetch a Work from the global state, and return the value via a promise.
+   *
+   * The resulting object returned on the resolved promis will be structured
+   * like so:
+   *
+   * ```
+   * {
+   *   "title": "String", // a unique title of the work; the natural key
+   *   "ISWC": "International Standard Work Code", // may be null
+   *   "songwriter_publisher_splits": [
+   *     {
+   *       // all splits in this list must add up to 100
+   *       "split": <a positive int>,
+   *       "songwriter_publisher": {
+   *         "songwriter_name": "IndividualIdentity.name",
+   *         "publisher_name": "OrganizationalIdentity.name" // must be Type.PUBLISHER
+   *       }
+   *     },
+   *     // and more
+   *   ]
+   * }
+   * ```
+   *
+   * @param {string} title - the title of the work (its natural key)
+   *
+   * @returns {Promise} the promise for the work value
+   *
+   * @since 1.0.0
+   */
+  getWork (title) {
+    return _omiStateEntry(this._sawtoothRestUrl, Work, title)
+  }
+
+  getWorks () {
+    return _omiStateCursor(this._sawtoothRestUrl, Work)
+  }
+}
+
+/**
+ * A BatchStatus object provides a function to query for the current batch
+ * status of a submitted transaction. That is, whether or not the batch has been
+ * committed to the block chain.
+ */
+class BatchStatus {
+  /**
+   * Constructs a BatchStatus for a given batch id, with a url for querying the
+   * current status.
+   *
+   * @param {string} batchId - the batchId
+   * @param {string} statusUrl - the batch status url
+   */
+  constructor (batchId, statusUrl) {
+    /**
+     * @member {string}
+     */
+    this.batchId = batchId
+    this._statusUrl = statusUrl
+  }
+
+  /**
+   * Returns the batch status from a transaction submission.
+   *
+   * The status is returned via a promise.  If a timeout occurs, the promise is
+   * rejected.
+   *
+   * @param {number} [timeout=300000] - a timeout in millis, defaults to 5
+   * minutes
+   *
+   * @returns {Promise<string>} the status
+   */
+  get (timeout = 300000) {
+    return Promise.race([
+      _promiseGet(this._statusUrl).then((body) => body.data[this.batchId]),
+      new Promise((resolve, reject) => {
+        setTimeout(() => reject(new Error('Timeout occurred')), timeout)
+      })
+    ])
+  }
+}
```

Comment:
No, it's not much better. But if 95% of the use is `if (status === 'COMMITTED')`, it would be a nice convenience to have `if (isCommitted)`. Saves the client a step, but also avoids them having to check for an arbitrary string value.